### ISSUE TITLE
Problem uploading string-vectors with NA

### ISSUE THREAD

**Issue by apoliakov** [Paradigm4/SciDBR#199]
```R
> x = as.character(c(NA,NA,NA))
> class(x)
[1] "character"
> x
[1] NA NA NA
> y = scidb::as.scidb(db, x)
Error: HTTP error 406
UserException in file: src/query/ops/input/InputArray.cpp function: moveNext line: 179
Error id: scidb::SCIDB_SE_IMPORT_ERROR::SCIDB_LE_FILE_IMPORT_FAILED
Error description: Import error. Import from file '/run/shm/shim_input_buf_H5nq15' (instance 0) to array 'tmp_input_array' failed at line 1, column 0, offset 5, value='(unreadable)': Failed to read file: 0.

> x = c('a','b','c')
> y = scidb::as.scidb(db, x)

> x = c('a','b',NA)
> y = scidb::as.scidb(db, x)
Error: HTTP error 406
UserException in file: src/query/ops/input/InputArray.cpp function: moveNext line: 179
Error id: scidb::SCIDB_SE_IMPORT_ERROR::SCIDB_LE_FILE_IMPORT_FAILED
Error description: Import error. Import from file '/run/shm/shim_input_buf_H5nq15' (instance 0) to array 'tmp_input_array' failed at line 3, column 2, offset 19, value='': Data exceeds array bounds (stepped out of bounds computing next chunk location).

> x = c('a','b',NA, 'd')
> y = scidb::as.scidb(db, x)
Error: HTTP error 406
UserException in file: src/query/ops/input/InputArray.cpp function: moveNext line: 179
Error id: scidb::SCIDB_SE_IMPORT_ERROR::SCIDB_LE_FILE_IMPORT_FAILED
Error description: Import error. Import from file '/run/shm/shim_input_buf_H5nq15' (instance 0) to array 'tmp_input_array' failed at line 4, column 0, offset 24, value='(unreadable)': Failed to read file: 0.
```

**Comment by ksens** [comment]
Just FYI -- works when you upload NA-s via a data.frame

```R
x = c('a','b',NA, 'd')
df1 = data.frame(x = x, stringsAsFactors = FALSE)
df1_scidb = scidb::as.scidb(db, df1)
identical(as.R(df1_scidb)$x, x)
# [1] TRUE
```

**Comment by apoliakov** [comment]
Yes - there are different upload paths - data frame uses text,
single-vector uses binary.
We were using the single-vector path because it is faster in some cases.
But it is vulnerable to this issue

**Comment by mcolosimo-p4** [comment]
This should be fixed in 63c67ac641a6fa29
